cristianino/prueba_innclod
# Feature/add documentation models

### Summary
This pull request introduces three new database migrations, Eloquent models, and their corresponding boilerplate for the following tables:

- `DOC_DOCUMENTO`: Represents documents within the system.
- `TIP_TIPO_DOC`: Categorizes types of documents.
- `PRO_PROCESO`: Defines processes that may be associated with the documents.

### Changes

#### Migrations
- Created migration for `doc_documentos` table with necessary fields such as `id`, `name`, `tip_tipo_doc_id`, `pro_proceso_id`, and timestamps.
- Created migration for `tip_tipo_docs` table including fields like `id`, `type_name`, and timestamps.
- Created migration for `pro_procesos` table with fields such as `id`, `process_name`, and timestamps.

#### Models
- Created `DocDocumento` model with relationships to `TipTipoDoc` and `ProProceso`.
- Created `TipTipoDoc` model with a relationship to `DocDocumento`.
- Created `ProProceso` model with a relationship to `DocDocumento`.

#### Relationships
- `DocDocumento` belongs to `TipTipoDoc` and `ProProceso`.
- `TipTipoDoc` has many `DocDocumento`.
- `ProProceso` has many `DocDocumento`.

#### Factories and Seeders
- Added factories for each model to facilitate testing and seeding of the database.
- Created seeders for each table to populate with initial data for development purposes.

### Testing
- Included tests to ensure that migrations run correctly.
- Added model tests to validate relationships and any custom methods.

### How to Test
- Pull the branch associated with this pull request.
- Run `php artisan migrate` to apply the database changes.
- Use the provided factories to generate dummy data for testing.
- Run `php artisan db:seed` to seed your database with initial data.
- Execute the test suite to validate that all aspects of the implementation work as expected.

### Additional Notes
- Ensure that your local environment is up to date with the latest changes from the `develop` or `main` branch before testing this pull request.
- Review the migrations to ensure that the fields align with the application's requirements.

Please review the changes and provide feedback or approval for the merge.

---

Remember to replace any placeholder text with the actual details from your pull request, such as field names and types, specific relationships (one-to-many, many-to-many, etc.), and any additional methods or properties added to the models.
